x-pack/test/functional/services/ml/stack_management_jobs.ts

```diff
@@ -274,23 +285,116 @@ export function MachineLearningStackManagementJobsProvider(
     },
 
     async importJobs() {
-      await retry.tryForTime(5000, async () => {
-        await testSubjects.click('mlJobMgmtImportImportButton', 1000);
+      await testSubjects.click('mlJobMgmtImportImportButton', 1000);
+      await testSubjects.missingOrFail('mlJobMgmtImportJobsFlyout', { timeout: 60 * 1000 });
+    },
+
+    async assertReadErrorCalloutExists() {
+      await testSubjects.existOrFail('~mlJobMgmtImportJobsFileReadErrorCallout');
+    },
+
+    async selectExportJobType(jobType: JobType) {
+      if (jobType === 'anomaly-detector') {
+        await testSubjects.click('mlJobMgmtExportJobsADTab');
+        await testSubjects.existOrFail('mlJobMgmtExportJobsADJobList');
+      } else {
+        await testSubjects.click('mlJobMgmtExportJobsDFATab');
+        await testSubjects.existOrFail('mlJobMgmtExportJobsDFAJobList');
+      }
+    },
+
+    async selectExportJobSelectAll() {
+      await testSubjects.click('mlJobMgmtExportJobsSelectAllButton');
+    },
+
+    async getDownload(filePath: string) {
+      return retry.tryForTime(5000, async () => {
+        expect(fs.existsSync(filePath)).to.be(true);
+        return fs.readFileSync(filePath).toString();
       });
-      await retry.tryForTime(40000, async () => {
-        await testSubjects.missingOrFail('mlJobMgmtImportJobsFlyout');
+    },
+
+    getExportedFile(fileName: string) {
+      return path.resolve(REPO_ROOT, `target/functional-tests/downloads/${fileName}.json`);
+    },
+
+    deleteExportedFiles(fileNames: string[]) {
+      fileNames.forEach((file) => {
+        try {
+          fs.unlinkSync(this.getExportedFile(file));
+        } catch (e) {
+          // it might not have been there to begin with
+        }
       });
     },
 
-    async assertImportedJobIdsExist(expectedJobIds: string[]) {
-      await retry.tryForTime(40000, async () => {
-        await testSubjects.click('mlRefreshJobListButton', 1000);
-        await testSubjects.missingOrFail('mlJobMgmtImportJobsFlyout');
+    async selectExportJobs() {
+      await testSubjects.click('mlJobMgmtExportExportButton');
```

Comment:
Would be good to validate that the click actually did something, e.g. validate that the success toast is displayed or that the flyout is closed.

Reply:
isn't the fact the file exported and exists on disk a good validation that the button worked?
The toast could still display but the actual export fail.

Reply:
Right, but this `exported file exists` check is not done in this service method, but called separately from the test file. We tried to make the service methods self-validating, so they can be used as individual building blocks.
Another point is: in some cases the UI automation runs through the steps so quickly that it e.g. sends a `click` to a button, but things haven't finished loading yet, so the click doesn't come through. With the current code, the next step (validating that the file exists) would fail, but that would make debugging harder as it wouldn't indicate the real issue (UI timing). Does that make sense?

Reply:
Updated in https://github.com/elastic/kibana/pull/110578/commits/62e901212a167e8db5384455296567e8b13963a8